token_update_transaction_e2e_test.go

```diff
@@ -843,3 +843,212 @@ func TestIntegrationTokenUpdateTransactionCannotUpdateImmutableNFT(t *testing.T)
+
+	resp, err := NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetAdminKey(env.OperatorKey).
+		SetSupplyKey(env.OperatorKey).
+		SetTokenType(TokenTypeNonFungibleUnique).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	tx, err := NewTokenUpdateTransaction().
+		SetTokenID(*receipt.TokenID).
+		SetAdminKey(NewKeyList()).
+		SetKeyVerificationMode(NO_VALIDATION).
+		FreezeWith(env.Client)
+
+	// Sign with the current admin key
+	resp, err = tx.Sign(env.OperatorKey).Execute(env.Client)
+	assert.NoError(t, err)
+
+	receipt, err = resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+}
+
+func TestIntegrationTokenUpdateTransactionUpdateAdminKeyWithoutAlreadySetAdminKeyFails(t *testing.T) {
+	t.Parallel()
+	env := NewIntegrationTestEnv(t)
+
+	zeroAdminKey, err := PrivateKeyFromString("0000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+
+	resp, err := NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTokenType(TokenTypeFungibleCommon).
+		SetInitialSupply(1000000).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// Cannot update token with no admin key
+	resp, err = NewTokenUpdateTransaction().
+		SetTokenID(*receipt.TokenID).
+		SetAdminKey(zeroAdminKey).
+		SetKeyVerificationMode(NO_VALIDATION).
+		Execute(env.Client)
+
+	assert.NoError(t, err)
+
+	receipt, err = resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.ErrorContains(t, err, "exceptional receipt status: TOKEN_IS_IMMUTABLE")
+}
+
+func TestIntegrationTokenUpdateTransactionUpdateAdminKeyWithoutSignFails(t *testing.T) {
+	t.Parallel()
+	env := NewIntegrationTestEnv(t)
+
+	zeroAdminKey, err := PrivateKeyFromString("0000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(t, err)
+
+	supplyKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	resp, err := NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetSupplyKey(supplyKey).
+		SetAdminKey(env.OperatorKey).
+		SetTokenType(TokenTypeNonFungibleUnique).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetFreezeDefault(false).
+		Execute(env.Client)
+	require.NoError(t, err)
+	receipt, err := resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	tx, err := NewTokenUpdateTransaction().
+		SetTokenID(*receipt.TokenID).
+		SetAdminKey(zeroAdminKey).
+		SetKeyVerificationMode(NO_VALIDATION).
+		FreezeWith(env.Client)
+	assert.NoError(t, err)
+
+	// Sign with the supply key should fail
```

Comment:
nit: If you have a comment here you can go ahead and add comments for the above steps as well. This would apply for other functions as well.

Reply:
Added comments